api/types/strslice/strslice_test.go

```diff
@@ -29,8 +29,8 @@ func TestStrSliceMarshalJSON(t *testing.T) {
 
 func TestStrSliceUnmarshalJSON(t *testing.T) {
 	parts := map[string][]string{
-		"":   {"default", "values"},
-		"[]": {},
+		"":                        {"default", "values"},
```

Comment:
As a workaround, we can add a blank line between the short and long keys to make it work for both 1.10 and 1.11 (see https://github.com/containerd/containerd/pull/2436) but it's really a workaround

I opened a ticket upstream for the `gofmt` change; https://github.com/golang/go/issues/26228

Reply:
Ok, this was marked as a "won't fix", "documentation only" change https://github.com/golang/go/issues/26228#issuecomment-402684430, so if we want to support multiple versions of Go, we need to make some changes to CI and/or the contributing guide;

> This change was done on purpose in https://github.com/golang/go/commit/542ea5ad91367d2b40589942cc5757a4d5f43f97.
> As to why gofmt changes every release - see @griesemer's comment in https://github.com/golang/go/issues/25161#issuecomment-385481599.
> However, I agree that this change should be documented in the release notes. I can't see it in https://tip.golang.org/doc/go1.11.

Reply:
Yes, this looks better (in most cases), thanks for suggestion!

Reply:
Discussing possible enhancements on the ticket upstream to ease migrating between Go versions.

Reply:
In most cases, if I remember correctly, once rebasing to a newer Go version we are losing compatibility with older ones either directly (i.e. some code in moby/moby starts using some feature from the new Go version) or indirectly (i.e. some code in vendored or otherwise used packages relies on a feature from the new Go version). I am about 84% sure current moby can't be compiled with Go 1.9, for example.

Therefore, chances are slim we'll be supporting Go 1.10 when Go 1.11 will become the current stable version of Go. Having said that, I am all for maintaining compatibility with older versions unless it requires some tremendous efforts.